R/infection_process.R

```diff
@@ -0,0 +1,48 @@
+#' @title Calculating the FOI
+#'
+#' @description calculating the FOI and infection process
+#'
+#' @param human the human handle
+#' @param states a list of states in the model
+#' @param discrete_age variable
+#' @param exposure event for covid exposure
+#' @noRd
+infection_process <- function(human, states, discrete_age, exposure) {
+
+  function(api) {
+    pars <- api$get_parameters()
+    inf_states <- api$get_state(human, states$IMild, states$ICase)
+
+    # If IMild = ICase = 0, FOI = 0, i.e. no infected individuals
+    if (length(inf_states) > 0) {
+
+      # Group infection by age
+      ages <- api$get_variable(human, discrete_age, inf_states)
+      inf_ages <- tabulate(ages, nbins = pars$N_age)
+
+      # Calculate FoI and use to create probability for each age group
+      m <- get_contact_matrix(pars$contact_matrix_set)
+      lambda <- pars$beta * rowSums(m %*% diag(inf_ages))
+
+      # Transition from S to E
+      susceptible <- api$get_state(human, states$S)
+      ages <- api$get_variable(human, discrete_age, susceptible)
+
+      # FOI for each susceptible person
+      lambda <- lambda[as.integer(ages)]
```

Comment:
If we are request `discrete_age` in line 29, is this `as.integer()` call needed?